src/js/angular/guides/steps/complex/ttyg/common/configure-agent/plugin.js

```diff
@@ -0,0 +1,68 @@
+PluginRegistry.add('guide.step', [
+    {
+        guideBlockName: 'configure-agent',
+        getSteps: (options, services) => {
+            const GuideUtils = services.GuideUtils;
+
+            const configureExtractionMethods = (services, options) => {
+                const methods = options.methods || [];
+
+                return methods.map((method) => {
+                    return {
+                        guideBlockName: method.guideBlockName,
+                        options: method.options
+                    };
+                });
+            };
+
+            return [
+                {
+                    guideBlockName: 'wait-for-element-to-show',
+                    options: angular.extend({}, {
+                        elementSelectorToShow: GuideUtils.getElementSelector('.agent-settings-modal'),
+                        timeToWait: 10
+                    }, options)
+                },
+                {
+                    guideBlockName: 'input-element',
+                    options: angular.extend({}, {
+                        content: 'guide.step_plugin.configure-agent.name-input',
+                        class: 'input-agent-name-guide-dialog',
+                        url: '/ttyg',
+                        beforeShowPromise: () => GuideUtils.waitFor(GuideUtils.getGuideElementSelector('agent-form'), 5)
+                            .catch((error) => {
+                                services.toastr.error(services.$translate.instant('guide.unexpected.error.message'));
+                                throw error;
+                            }),
+                        elementSelector: GuideUtils.getGuideElementSelector('agent-name'),
+                        showOn: () => options.name,
```

Comment:
Refactor the method 'getSteps' to conditionally add this step if the option name ('editName') is passed, and remove 'showOn'. There's no need to add the steps and skip them at runtime; we can conditionally add them at 'compile' time. The same applies to other steps that depend on the option and not on the system state.